src/main/java/dk/cngroup/trainings/spring/springassignment/controller/web/AnimalWebController.java

```diff
@@ -1,19 +1,17 @@
 package dk.cngroup.trainings.spring.springassignment.controller.web;
 
+import dk.cngroup.trainings.spring.springassignment.exception.AnimalNotFoundException;
 import dk.cngroup.trainings.spring.springassignment.exception.InvalidAnimalException;
 import dk.cngroup.trainings.spring.springassignment.model.Animal;
 import dk.cngroup.trainings.spring.springassignment.service.AnimalService;
 import org.springframework.stereotype.Controller;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.ModelAttribute;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.*;
 import org.springframework.web.servlet.ModelAndView;
 
-import javax.validation.Valid;
+import java.util.List;
 
 @Controller
-@RequestMapping("/web/animals")
+@RequestMapping("/web/animal")
 public class AnimalWebController {
 
 	private AnimalService animalService;
```

Comment:
`animalService` can be `final` here. It is a good practise to have dependencies as final, so nobody can switch them once a class is instantiated (applies to other parts of app too)

Reply:
Nice touch.